src/Core/Checkout/Document/Renderer/ZugferdRenderer.php

```diff
@@ -94,6 +94,9 @@ protected function createDocument(RendererResult $renderResult, OrderEntity $ord
         // So no A11y will be generated
         $config->merge(['fileTypes' => ['xml']]);
 
+        // create version of order to ensure the document stays the same even if the order changes
+        $operation->setOrderVersionId($this->orderRepository->createVersion($order->getId(), $context, 'document'));
```

Comment:
this seems to me that it is set only on the PHP object, but never persisted, right? We would need the FK of document to also use the non-live-version. Same with DeliveryNoteRenderer.

Reply:
Never mind, the persistence seems to be done by the `DocumentGenerator` :)